homeassistant/components/formula_one/__init__.py

```diff
@@ -0,0 +1,289 @@
+            ):
+                return True
+
+        for new_standing in new_driver_standings:
+            if (
+                new_standing.position not in cur_driver_positions
+                or new_standing.driver.driver_id not in cur_driver_ids
+            ):
+                return True
+
+        if len(new_races) != cur_race_count:
+            return True
+
+        return False
+
+    def get_constructor_standing_by_id(
+        self, constructor_id: str
+    ) -> ergast.ConstructorStanding | None:
+        """Get ConstructorStanding for a constructor_id."""
+        for standing in self.constructor_standings:
+            if standing.constructor.constructor_id == constructor_id:
+                return standing
+
+        return None
+
+    def get_constructor_standing_by_position(
+        self, position: int
+    ) -> ergast.ConstructorStanding | None:
+        """Get ConstructorStanding for a position."""
+        for standing in self.constructor_standings:
+            if standing.position == position:
+                return standing
+
+        return None
+
+    def get_driver_standing_by_id(self, driver_id: str) -> ergast.DriverStanding | None:
+        """Get DriverStanding for a driver_id."""
+        for standing in self.driver_standings:
+            if standing.driver.driver_id == driver_id:
+                return standing
+
+        return None
+
+    def get_driver_standing_by_position(
+        self, position: int
+    ) -> ergast.DriverStanding | None:
+        """Get DriverStanding for a position."""
+        for standing in self.driver_standings:
+            if standing.position == position:
+                return standing
+
+        return None
+
+    def get_race_by_round(self, round_no: int) -> ergast.Race | None:
+        """Get Race for round."""
+        for race in self.races:
+            if race.round_no == round_no:
+                return race
+
+        return None
+
+    def get_next_race(self) -> ergast.Race | None:
+        """Get the next Race."""
+        now = datetime.utcnow().replace(tzinfo=pytz.UTC)
+
+        for race in self.races:
+            race_date = race.date.replace(tzinfo=pytz.UTC)
+            if race_date > now:
+                return race
+
+        for race in self.races_next_year:
+            race_date = race.date.replace(tzinfo=pytz.UTC)
+            if race_date > now:
+                return race
+
+        return None
+
+    async def test_connect(self) -> bool:
+        """Test if we can connect with the host."""
+        try:
+            await self.hass.async_add_executor_job(
+                ergast.Ergast().season().get_constructor_standings
+            )
+        except Exception as ex:  # pylint: disable=broad-except
+            _LOGGER.error("Test connection failed: %s", ex)
+            return False
+
+        return True
+
+
+class F1UpdateCoordinator(DataUpdateCoordinator):
```

Comment:
The data it holds is missing, right now, it is `None`, but it should hold the actual data (as mentioned above too).

```suggestion
class F1UpdateCoordinator(DataUpdateCoordinator[None]):
```